api/odl.py

```diff
@@ -570,6 +831,32 @@ def process_one_page(self, response):
 
         return next_url
 
+class ODLHoldReaper(CollectionMonitor):
+    """Check for holds that have expired and delete them, and update
+    the holds queues for their pools."""
+
+    SERVICE_NAME = "ODL Hold Reaper"
+    PROTOCOL = ODLWithConsolidatedCopiesAPI.NAME
+
+    def __init__(self, _db, collection=None, api=None, **kwargs):
+        super(ODLHoldReaper, self).__init__(_db, collection, **kwargs)
+        self.api = api or ODLWithConsolidatedCopiesAPI(_db, collection)
+
+    def run_once(self, start, cutoff):
+        # Find holds that have expired.
+        expired_holds = self._db.query(Hold).join(
+            Hold.license_pool
+        ).filter(
+            LicensePool.collection_id==self.api.collection_id
+        ).filter(
+            Hold.end<datetime.datetime.utcnow()
+        )
+
+        for hold in expired_holds:
+            pool = hold.license_pool
+            self._db.delete(hold)
+            self.api.update_hold_queue(pool)
```

Comment:
You can make this more efficient by keeping track of the LicensePools that need to be updated, and then calling `update_hold_queue` on each pool in a separate loop. Since holds are concentrated in a few very popular books, it's pretty likely that the same LicensePool will show up more than once here.